### PR TITLE
BasicDataStructuresMapTestSuite reworked

### LINKED ISSUE #103: Refactor: Rework BasicDataStructuresMapTestSuite to be more in-line with newer tests
### Description

BasicDataStructuresMapTestSuite contains 7 facts at the moment, with multiple asserts per fact. This should be broken up into more facts to be more thorough.

### Script Name

BasicDataStructuresMapTestSuite 

### Steps To The Task

- [x] Split up all tests within existing facts into their own facts
- [x] Add comments
- [x] Update assert descriptions

### PR DESCRIPTION
**Pull request by AdamBorek**
Facts have been broken up into multiple smaller ones

### CODE REVIEW

File: projects/xUnit/scripts/BasicDataStructuresMapTestSuite/BasicDataStructuresMapTestSuite.gml [hunk 1]
```diff
@@ -49,41 +91,111 @@ function BasicDataStructuresMapTestSuite() : TestSuite() constructor {
...
-	})
-
-	addFact("ds_map_read_write_test", function() {
-
-		//SB: ds_map_copy, ds_map_delete, ds_map_write, ds_map_clear, ds_map_empty and ds_map_read
+		var _map, _output;
+		
+		_map = ds_map_create();
+		
+		_output = ds_map_empty(_map)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataStructuresMapTestSuite/BasicDataStructuresMapTestSuite.gml]:
```suggestion
		_output = ds_map_empty(_map);
```

File: projects/xUnit/scripts/BasicDataStructuresMapTestSuite/BasicDataStructuresMapTestSuite.gml [hunk 2]
```diff
@@ -49,41 +91,111 @@ function BasicDataStructuresMapTestSuite() : TestSuite() constructor {
...
+			
+		var _map, _output;
+		
+		_map = ds_map_create();
+			
+		ds_map_add_map(_map, "Map", ds_map_create());
 		
-		ds_map_destroy(0);
-		ds_map_destroy(1);
+		_output = ds_map_empty(_map)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataStructuresMapTestSuite/BasicDataStructuresMapTestSuite.gml]:
```suggestion
		_output = ds_map_empty(_map);
```

File: projects/xUnit/scripts/BasicDataStructuresMapTestSuite/BasicDataStructuresMapTestSuite.gml [hunk 3]
```diff
@@ -182,17 +314,19 @@ function BasicDataStructuresMapTestSuite() : TestSuite() constructor {
 		_map = ds_map_secure_load_buffer(_mapSaveBuffer);
 		
 		_output = ds_map_empty(_map);
-		assert_false(_output, "#1 ds_map_secure_load_buffer(), loaded empty map (could be a ds_map_secure_save_buffer problem)");
-		assert_map_size(_map, 1, "#1 ds_map_secure_load_buffer(), loaded wrong number of elements (could be a ds_map_secure_save_buffer problem)");
+		assert_false(_output, "ds_map_secure_load_buffer(), loaded empty map (could be a ds_map_secure_save_buffer problem)");
+		assert_map_size(_map, 1, "ds_map_secure_load_buffer(), loaded wrong number of elements (could be a ds_map_secure_save_buffer problem)");
 		_output = ds_map_exists(_map, "Three")
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataStructuresMapTestSuite/BasicDataStructuresMapTestSuite.gml]:
```suggestion
		_output = ds_map_exists(_map, "Three");
```

**Review by Indibulous — APPROVED** [review]
Seems good! just added a few missing semicolons.

**Review by DiasFranciscoA — APPROVED** [review]

**Pull request merged**